SEED-platform/seed
# Disable ability to remove matching criteria fields once inventory is added

#### Any background context you want to provide?
An organization has a list of columns that are used as matching criteria. When new properties are imported into a cycle, SEED will check the imported property's matching columns against all cycle properties matching columns. If all criteria are met (`None`s excluded) the properties will be merged. If there is a partial match the imported property will be treated as unique and not merged. 

- If matching criteria are added then there is no risk of existing properties being merged as merges require a complete match.
- If matching criteria are removed then existing properties within a cycle can be merged. This is problematic as properties with different Access Level Instances can be merged. This undermines the permission tree as child users should not be allowed to change parent user properties.

#### What's this PR do?
Disables the ability for a user to remove matching criteria once an organization has inventory (users can always add matching criteria), and removes `match_merge_link` endpoints

#### How should this be manually tested?
- Start with 0 inventory 
- Go to column settings and add/remove matching criteria
- Import properties 
- Go to column settings. You should only be able to add matching criteria. Existing criteria should be gray checkboxes

#### What are the relevant tickets?
#4219 

#### Screenshots (if appropriate)
